setup-ruby-and-rust/action.yml

```diff
@@ -349,23 +347,11 @@ runs:
           exit 1
         fi
 
-        # Add stdckdint.h from LLVM 18 to prevent Ruby 3.4's header from erroring
-        extra_includes=$msys_root/opt/rb-sys/include
-        mkdir -p $msys_root/opt/rb-sys/include
-        cp 'C:\Program Files\LLVM\lib\clang\18\include\stdckdint.h' $extra_includes
+        bindgen_extra_clang_args="--target=${{ steps.derive-toolchain.outputs.toolchain }} --sysroot=$msys_root -D__AVX512VLFP16INTRIN_H -D__AVX512FP16INTRIN_H"
```

Comment:
@jbourassa @ianks I might be too optimistic, but did this just work? 

- This build: https://github.com/stanhu/actions/actions/runs/12956772391
- Previously failing build: https://github.com/jbourassa/oxidize-rb-actions/actions/runs/12583936454/job/35072741265
- Reference: https://github.com/rust-lang/rust-bindgen/issues/2500#issuecomment-1640545912

Reply:
Looks like it did! Not sure what the exact implications of defining those headers are, but it sounds reasonable to me.

I'm willing to merge this, but I'll wait a bit to see if Ian has something to add.

Reply:
https://github.com/oxidize-rb/oxi-test/pull/13 fixes the `ruby-head` errors.

Reply:
I think defining those preprocessor headers effectively disables the ability to use of [AVX-512 FP16 intrinsics](https://www.intel.com/content/www/us/en/docs/cpp-compiler/developer-guide-reference/2021-9/intrinsics-for-avx-512-instructions.html).

I see that these intrinsics were added in clang 14 (https://github.com/llvm/llvm-project/commit/6f7f5b54c81be59ec7876649d1f9aa6b104658ec), but in clang 16 these intrinsics may have been included by default if SSE2 is available: https://github.com/llvm/llvm-project/commit/08388ad81e9d65116b0857b0ebc9bcf7f2be5e31.